### DISCUSSION TITLE
Can pygeoapi be used as a proxy for other OGC Feature services?

### DISCUSSION THREAD

**Discussion by geographika** [geopython/pygeoapi#1327] [Q&A] [answered]
Hi all,

Is there any way to configure pygeoapi to be used as a proxy to other OGC Feature services?
The example use case would be if you had a MapServer instance configured to serve OGC Feature services and wanted to serve this out with other services as part of a single pygeoapi instance. The advantage would be it would be part of a feature catalog at a single URL. 
I'm guessing it would be a type of proxy provider similar to the ESRIServiceProvider. 
Hopefully this question makes sense, and thanks for any input. 

Seth

**Comment by justb4** [comment, marked as answer]
Yes!: for proxying WFS the OGR Provider can be configured, and for proxying WMS the pygeoapi Maps Provider has means for both WMS and (MapScript) MapServer. 

See docs: 
* https://docs.pygeoapi.io/en/latest/data-publishing/ogcapi-features.html
* https://docs.pygeoapi.io/en/latest/data-publishing/ogcapi-maps.html

There is even content in the recent workshop at FOSS4G:

* WFS: https://dive.pygeoapi.io/publishing/ogcapi-features/#pygeoapi-as-a-wfs-proxy
* WMS: https://dive.pygeoapi.io/publishing/ogcapi-maps/#serve-a-wms-via-ogc-api-maps

**Reply by geographika** [reply]
Thanks @justb4. For older versions of MapServer using WFS and the OGR Provider is perfect.
I was more wondering if the OGC Features API added in MapServer 8.0 could be read through pygeoapi - https://www.mapserver.org/ogc/ogc_api.html

Looking at the OGR Provider it supports OGC Features using the OAPIF driver so maybe this would be the way to go?
https://gdal.org/drivers/vector/oapif.html#vector-oapif

**Reply by justb4** [reply]
@geographika ok, I see. Yes, that would be the only way currently supported by pygeoapi, we don't have a OAPIF proxy (yet). Indeed the OGR Provider with OAPIF Driver should in theory work, but has not yet been tried...Let us know!
